aten/src/TH/generic/THTensor.cpp

```diff
@@ -432,13 +432,15 @@ void THTensor_(select)(THTensor *self, THTensor *src, int dimension, int64_t sli
   if(!src)
     src = self;
 
+#ifndef USE_TH_SCALAR
   THArgCheck(src->_dim() > 1, 1, "cannot select on a vector");
```

Comment:
I don't follow why this check applies when `!USE_TH_SCALAR`. Just because I'm `USE_TH_SCALAR` doesn't mean I can select on a vector... does it?

Reply:
I'm assuming the check was originally there because the proper thing to do is to return a scalar.

Reply:
Makes sense! Excellent.